### DISCUSSION TITLE
Replacement of CCU2 by CCU3 - any known issues?

### DISCUSSION THREAD

**Discussion by moelloman** [SukramJ/aiohomematic#1422] [Q&A] [answered]
Hi all,
I've running a stable integration of a CCU2 with 65 devices (2 LAN gateways, door and window sensors, sockets, smoke detectors, switches etc.). I plan to replace the CCU2 by a CCU3. I already did an export from the CCU2 and imported it on the CCU3 (isolated network ), everything looks good from HM perspective. Next step would be shutting down the CCU2 and plug in the CCU3 instead (same IP). Then probably a restart on HA or reload of the integration.
I'm wondering wether the replacement from the HomematicIP integration perspective will be "accepted" and all devices & entities will be kept without any issues or do I have to remove all devices from the CCU2 and let discover the same but from the CCU3 now or what else? What is the "migration path"?

My current config:
HA core 2024.1.4
HA OS: 11.5
Homematic(IP) local: 1.54.0
CCU2/3: both on latest FW

Many thanks for your help in advance!

**Comment by moelloman** [comment, marked as answer]
For all who are interested in:
I did the replacement successfully. Those are the steps:
- update FW on CCU2 and CCU3 to the latest versions
- backup CCU2
- deactivate Homematic IP intetgration on HA
- shutdown CCU2
- Initial installation steps on CCU3
- on CCU3 restore the backup file from CCU2
- after reboot CCU3 make sure:  IP settings are the same as on CCU2, FW settings fot XML-RPC API are on "full access", evtl. the same for script API
- change the name of the CCU3 (if you like)
- check on CCU3 whether all devices are connected and responsive (incl. LAN Gateways, if existing)
- on HA activate Homematic IP integration (activate diagnostic logging)
- on HA check all Homematic devices (for trouble shootting use the log file)
- finally reboot HA, all device status should be updated
- rename the device name CCU2 to CCU3 (if you like)

Finally I can say the replacement of a CCU2 by a more modern CCU3 works easy and smoothly.  👍

**Reply by ivo7B4** [reply]
Thank you! How did you change the name?
"change the name of the CCU3 (if you like)"
I ask as I just see "ccu3-webui" (in my case) without any option to edit.

I actually have a similar plan. I just did a question as it still looks different as the migration that you have done. But your description is the closest that I've found to what I plan to do (change from a CCU3 to a Rasperrymatic running on Proxmox).
I also don't think that I make sure that my new device gets the same static IP as my previous device. But this I think should be possible as the IP address can be changed with the configure button.

**Reply by moelloman** [reply]
Hi ivo7B4 ,
in the integration the CCU itself is listed as a device. It was named CCU2 (coming from the old one, because I use the same IP etc.). In the device properties I just changed the name by clicking on the pen symbol in the upper right corner. 
![image](https://github.com/danielperna84/hahomematic/assets/81770471/1e142fad-e2ed-43b7-bdec-80273245bdb8)
